software/database/src/main/java/brooklyn/entity/database/postgresql/PostgreSqlSshDriver.java

```diff
@@ -307,6 +327,17 @@ public void customize() {
         // on port 5432?" error then the port is probably closed. Check that the firewall allows external TCP/IP
         // connections (netstat -nap). You can open a port with lokkit or by configuring the iptables.
     }
+    
+    protected String getUserPassword() {
+        String password = entity.getConfig(PostgreSqlNode.PASSWORD);
+        if (Strings.isEmpty(password)) {
+            log.debug(entity + " has no password specified for " + PostgreSqlNode.PASSWORD + "; using a random string");
+            password = brooklyn.util.text.Strings.makeRandomId(8);
+            entity.setAttribute(PostgreSqlNode.PASSWORD, password);
```

Comment:
Why set the attribute too? It would be quite confusing to `setAttribute(password)` but have it overwritten when `getUserPassword` is called. Perhaps `USERNAME` and `PASSWORD` are just `ConfigKeys` that are set for you if not given?